File: BitManipulation/268. 丢失的数字/268. 丢失的数字.cpp
// 方法1 : 複製一組由0~len(nums)的list，並與原list相減
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int res = nums.size();//最後一個數字
        for(int i = 0; i < nums.size();i++){
            res += (i - nums[i]);
        }
        return res;
    }
};

// 位運算XOR
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int res = nums.size();
        for(int i = 0; i < nums.size(); i++){
            res ^= i^ (nums[i]);
        }

        return res;
    }
};

File: BitManipulation/268. 丢失的数字/268. 丢失的数字.py
'''
方法2:透過XOR，res ^= i ^ nums[i]
ex:[3,0,1]-->res = 3^(0^3)^(1^0)^(2^1) = (3^3)^(1^1)^(0^0)^2 = 2
'''

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        res = len(nums)
        for i in range(len(nums)):
            res ^= i ^ nums[i]
        return res